pkg/operator/ceph/nfs/controller.go

```diff
@@ -204,6 +205,15 @@ func (r *ReconcileCephNFS) reconcile(request reconcile.Request) (reconcile.Resul
 	}
 	r.clusterInfo.CephVersion = currentCephVersion
 
+	if r.clusterInfo.CephVersion.IsAtLeastPacific() {
+		if r.clusterInfo.CephVersion.IsAtLeast(version.CephVersion{Major: 16, Minor: 2, Extra: 6}) {
+			cephNFS.Spec.RADOS.Pool = dotNFSPoolName
+		} else {
+			cephNFS.Spec.RADOS.Pool = nfsGaneshaPoolName
```

Comment:
Prior to 16.2.6 didn't the user have to set this in the CR, so we shouldn't overwrite it?

Reply:
From what I can tell looking at the Ceph code,prior to 16.2.5 there was also a default pool name and the name was "nfs-ganesha" set [here](https://github.com/ceph/ceph/blob/pacific/src/pybind/mgr/nfs/utils.py#L3) and used [here](https://github.com/ceph/ceph/blob/pacific/src/pybind/mgr/nfs/export.py), so it looks like Octopus is the version that allows customizing the pool name and namespace.

Reply:
Your PR currently doesn't allow it to be set for octopus either. Seems like we can remove the `else` block to keep the current behavior that they set the pool name in the CR if older than 16.2.6

Reply:
So the way I see it: 
- Octopus: Customization is allowed, so don't change the pool and namespace
- Pacific before 16.2.6: No customization, default pool name is nfs-ganesha
- Pacific after 16.2.6: No customization, default pool name is .nfs
This code is supposed to change the pool and namespace if the version is pacific, if its above 16.2.6 it overwrites it to ".nfs", else it changes it to "nfs-ganesha". If the version precedes Pacific it doesn't change it at all and it should be whatever the user provided in the spec.

Reply:
Ok I see now that the pre-pacific settings won't be overwritten. Thanks for the clarification.

Reply:
I think @josephsawaya last comment is valuable and should be a comment before this conditional block.